src/rust/src/backend/ciphers.rs

```diff
@@ -183,14 +183,18 @@ impl CipherContext {
             // SAFETY: We ensure that outbuf is sufficiently large above.
             unsafe {
                 let n = if self.py_mode.bind(py).is_instance(&types::XTS.get(py)?)? {
```

Comment:
Actually, we should hoist this out of the loop, and then you can wrpa the whole loop in `py.allow_threads()`. This will improve readability, and also means we won't re-acquire the GIL multiple times for a giant ciphertext.

Reply:
Great point, just updated this.